pennylane_qiskit/qiskit_device.py

```diff
@@ -154,25 +162,26 @@ def __init__(self, wires, provider, backend, shots=1024, **kwargs):
 
             self.shots = 1024
 
-        self._backend = None
-
         self.provider = provider
-        self.backend_name = backend
-
-        def _get_backend_name(name):
-            return name if isinstance(name, str) else name()
 
-        self._capabilities["backend"] = [
```

Comment:
tbh I'm not entirely sure why this was added to the device capabilities, but again, users might depend on it. This PR could be considered "breaking" in that sense, but really this should have been private even in the old implementation